source/windows/clock.c

```diff
@@ -0,0 +1,49 @@
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include <aws/common/clock.h>
+#include <Windows.h>
+
+static const LONGLONG MUS_PER_SEC = 1000000;
+static const LONGLONG NS_PER_MUS = 1000;
+static const LONGLONG FILE_TIME_TO_NS = 100;
+static const LONGLONG EC_TO_UNIX_EPOCH = 11644473600LL;
```

Comment:
What's EC? add a comment?
Why use LONGLONG instead of the more explicit int32_t or int64_t?

Reply:
Windows has a different epoch than linux epoch. That's the conversion number.

LONGLONG is what's in the windows api reference and "technically" LONGLONG is defined with a precise width (only a minimum)